### ISSUE TITLE
GPUMPS Lassen experience

### ISSUE THREAD

**Issue by mtitov** [radical-cybertools/radical.pilot#2151]
While having tests of CVD WF2 at Lassen there was the following behavior of using GPUs:

4 tasks (1 cpu process, 4 cpu threads, 1 gpu process) were assigned to 1 node:
- every task had the Resource Set file (`unit.00000x.rs`) used at the option `--erf_input`, examples of that files content are the following:
```
rank: 0: { host: 1; cpu: {0,1,2,3}; gpu: {0}}
```
```
rank: 0: { host: 1; cpu: {12,13,14,15}; gpu: {3}}
```
- for every task was set the following env variable `export "CUDA_VISIBLE_DEVICES=0"` ([jsrun cvd_id_mode](https://github.com/radical-cybertools/radical.pilot/blob/devel/src/radical/pilot/agent/launch_method/jsrun.py#L34) affects the procedure setting `CUDA_VISIBLE_DEVICES ` [here](https://github.com/radical-cybertools/radical.pilot/blob/devel/src/radical/pilot/agent/scheduler/base.py#L851))
- As the result all 4 tasks were running on `GPU 0`

With the latest commit 21e92d1 (for Lassen) it was forced to assign different values at `CUDA_VISIBLE_DEVICES`
- As result each task was running on a different GPUs

---

Lassen uses `LSF` as a batch scheduler and `jsrun` as a command to launch tasks. The line `#BSUB -alloc_flags 'gpumps smt4'` is set.

**Comment by andre-merzky** [comment]
I think I actually know what happened here: we changed the way GPU and Core indexing works in the resource set for jsrun, and that changes the way GPUs are numbered, so that RP's `CUDA_VISIBLE_DEVICES` setting is off.   At least that's my guess...  Can you please check what `CUDA_VISIBLE_DEVICES` was set to in the task scripts for the suspicious runs?

**Comment by mtitov** [comment]
For the 1st run for every task it was `export "CUDA_VISIBLE_DEVICES=0"` (that was inside each shell script) - and I checked that all of them were at `GPU 0` using `nvidia-smi` at the compute node. For the 2nd run (after that commit) -> `export "CUDA_VISIBLE_DEVICES=0"`, `export "CUDA_VISIBLE_DEVICES=1"`, etc. and `nvidia-smi` showed each task had its own gpu

**Comment by andre-merzky** [comment]
Have a look [here](https://github.com/radical-cybertools/radical.pilot/blob/devel/src/radical/pilot/agent/scheduler/base.py#L849) please - does that address the issue?  Note that the last part of the initial comment seems outdated: physical indexing is not hardcoded anymore, but can be toggled via
```
        lm_info     = self._cfg['rm_info']['lm_info']
        cvd_id_mode = lm_info.get('cvd_id_mode', 'physical')
```

**Comment by mtitov** [comment]
To start NVIDIA MPS (multi-process service) on Sierra systems - provided by support team (references: [MPS](https://lc.llnl.gov/confluence/display/SIERRA/MPS), [MPS announcement as of June 18 2020](https://lc.llnl.gov/confluence/display/CORALEA/Announcement+Archive)):
> run inside an allocation:
> 
> **enable_mps**
> 
> to start the daemon on every node. This essentially does 'nvidia-cuda-mps-control -d' on every compute node with error checking via jsrun.
> 
> The CORAL epilogues should tear down the MPS daemons at the end of your job but they can also be torn down with:
> 
> **disable_mps**
> 
> This essentially does 'echo "quit" -t 0 | nvidia-cuda-mps-control' on every compute node with error checking via jsrun.

This ticket is closed in favor of #2140
